sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -458,6 +458,7 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
             // [COUNT(DISTINCT bar), COUNT(DISTINCT foo)] is disallowed because those two distinct
             // aggregates have different column expressions.
             val distinctExpressions = functionsWithDistinct.head.aggregateFunction.children
+              .filterNot(_.foldable)
```

Comment:
I tried `select count(distinct 1) from v group by id` but works fine, I think `foldable` is not the real problem here. I think the root cause is, FIRST/LAST put `ignoreNulls` as its children while the children are supposed to be the function inputs.

How about this fix
```
case class First(child: Expression, ignoreNulls: Boolean)
  extends DeclarativeAggregate with ExpectsInputTypes {

  def this(child: Expression) = this(child, false)

  def this(child: Expression, ignoreNullsExpr: Expression) = {
    this(child, First.validateIgnoreNullExpr(ignoreNullsExpr)) // follow HyperLogLogPlusPlus.validateDoubleLiteral
  }
```

Reply:
Ah, looks okay. I'll update.